### DISCUSSION TITLE
running F` inside a docker container

### DISCUSSION THREAD

**Discussion by shalevm** [nasa/fprime#1737] [Q&A] [answered]
I have an fprime project running inside a docker container. 
The app itself runs fine inside the container, the problem is that when I specify the default communication interface port in the run command, namely:
`./build-artifacts/Linux/bin/my_app_name -a 0.0.0.0 -p 50000`
in order to connect to it from the host with fprime-gds, and run the image in a container with a mapped port, namely:
`docker run -p 50000:50000 my_image`
fprime fails to create the socket and continuously prints out
`[WARNING] Failed to open port with status -4 and errno 111`
(line 69 from SocketReadTask.cpp). 
Just to clarify - this happens even before I start running frpime-gds on the host.
I should mention that outside a docker this process of connecting manually from gds to the flight software works perfectly fine,
which brings me to the conclusion that the problem is not inherently in fprime (or my code), but either in the way I'm using docker or the interaction between fprime and the container.
Does anyone have any idea what might be the problem?
Thanks.

**Comment by shalevm** [comment]
update: the failure comes from the call to `::connect` in `TcpClientSocket::openProtocol`.
the requested socket details: 0.0.0.0:50000 (as expected).
any suggestions?

**Comment by capsulecorplab** [comment]
I'm also running fprime in a docker container, but built on top of the kasm virtual desktop stack. I believe it was building and running the fprime/Ref app on v3.1.1 just fine. You're welcome to check it out here https://github.com/capsulecorplab/kasm-fprime-workspace

**Reply by shalevm** [reply]
Thanks Sean, but I need to use my own image, which, like I said, is running just fine, the problem is opening a tcp socket.

**Comment by shalevm** [comment, marked as answer]
Don't ask me why, but adding -it to the docker command fixed it...
